app/assets/javascripts/_page_show.es6

```diff
@@ -58,11 +58,66 @@
       });
     }
 
+    function isImageLink(link){
+        return link.find("img").length > 0
+    };
+
+    function isButton(link){
+        return link.hasClass("usa-button") || link.is("[class*='btn']") || link.is("[class*='button']");
+    }
+
+    function isIcon(link){
+        return link.children("i").length > 0;
+    }
+
+    function isPracticeLink(link){
+        return link.hasClass("dm-practice-link");
+    }
```

Comment:
Instead of having all of these conditions, I was thinking we could add a class to the container div of each `PageParagraphComponent`, and then just check if the closest `div` has that class. So something like:

`page_show.html.erb`
```
<%
  component = pc.component_type.constantize.find(pc.component_id)
  last_component = @page.page_components.size - 1
  is_paragraph_component = pc.component_type == 'PageParagraphComponent'
%>

<%# Page Header and Paragraph %>
<% if (pc.component_type == 'PageHeaderComponent' || is_paragraph_component) && component.text.present? %>
  <div
    class="margin-bottom-4-important <%= page_narrow_classes %><%= ' margin-bottom-0' if last_component === index %><%= 'paragraph-component' if is_paragraph_component %>"
  >
    <%= component.text.html_safe %>
  </div>
<% end %>
```

`_page_show.es6`
```
if (currentLink.closest('div').hasClass('paragraph-component')){
    currentLink.addClass("usa-link");
}
```

Reply:
ok! I implemented something pretty close to this so that accordion components are also included.